mempool/clist_mempool.go

```diff
@@ -306,14 +307,16 @@ func (mem *CListMempool) globalCb(req *abci.Request, res *abci.Response) {
 		checkType := req.GetCheckTx().GetType()
 		switch checkType {
 		case abci.CHECK_TX_TYPE_CHECK:
-			if mem.recheckCursor != nil {
+			if !mem.recheck.done() {
 				// this should never happen
 				panic("recheck cursor is not nil before resCbFirstTime")
 			}
 			mem.resCbFirstTime(req.GetCheckTx().Tx, res.GetCheckTx())
 
 		case abci.CHECK_TX_TYPE_RECHECK:
-			if mem.recheckCursor == nil {
+			tx := types.Tx(req.GetCheckTx().Tx)
```

Comment:
This is only used in case of error. We can move it to inside the next if clause.

Reply:
Done.